packages/client/src/__tests__/createErrorMessageWithContext.test.ts

```diff
@@ -0,0 +1,381 @@
+    getLocation() {
+      return { fileName, lineNumber, columnNumber }
+    },
+  }
+}
+
+function mockFile(mockedPath: string, content: string) {
+  vol.fromJSON({
+    [mockedPath]: content,
+  })
+}
+
+afterEach(() => vol.reset())
+
+test('basic', () => {
+  expect(
+    createErrorMessageWithContext({
+      originalMethod: 'model.findFirst',
+      callsite: mockCallsite('/project/some-file.js', 12, 34),
+      message: 'What a terrible failure!',
+    }),
+  ).toMatchInlineSnapshot(`
+
+    Invalid \`prisma.model.findFirst()\` invocation:
+
+
+    What a terrible failure!
+  `)
+})
+
+test('basic panic', () => {
+  expect(
+    createErrorMessageWithContext({
+      originalMethod: 'model.findFirst',
+      callsite: mockCallsite('/project/some-file.js', 12, 34),
+      isPanic: true,
+      message: 'What a terrible failure!',
+    }),
+  ).toMatchInlineSnapshot(`
+
+    Oops, an unknown error occured! This is on us, you did nothing wrong.
+    It occured in the \`prisma.model.findFirst()\` invocation:
+
+
+    What a terrible failure!
+  `)
+})
+
+test('without callsite', () => {
+  expect(
+    createErrorMessageWithContext({
+      originalMethod: 'model.findFirst',
+      message: 'What a terrible failure!',
+    }),
+  ).toMatchInlineSnapshot(`
+
+    Invalid \`prisma.model.findFirst()\` invocation:
+
+
+    What a terrible failure!
+  `)
+})
+
+test('with matching source file', () => {
+  mockFile('/project/some-file.js', 'prisma.model.findFirst({})')
+  expect(
+    createErrorMessageWithContext({
+      originalMethod: 'model.findFirst',
+      callsite: mockCallsite('/project/some-file.js', 1, 1),
+      message: 'What a terrible failure!',
+    }),
+  ).toMatchInlineSnapshot(`
+
+    Invalid \`prisma.model.findFirst()\` invocation in
+    /project/some-file.js:1:1
+
+    → 1 prisma.model.findFirst(
+
+    What a terrible failure!
+  `)
+})
+
+test('panic with matching source file', () => {
+  mockFile('/project/some-file.js', 'prisma.model.findFirst({})')
+  expect(
+    createErrorMessageWithContext({
+      originalMethod: 'model.findFirst',
+      callsite: mockCallsite('/project/some-file.js', 1, 1),
+      isPanic: true,
+      message: 'What a terrible failure!',
+    }),
+  ).toMatchInlineSnapshot(`
+
+    Oops, an unknown error occured! This is on us, you did nothing wrong.
+    It occured in the \`prisma.model.findFirst()\` invocation in
+    /project/some-file.js:1:1
+
+    → 1 prisma.model.findFirst({})
+
+    What a terrible failure!
+  `)
+})
+
+test('with matching source file, but without matching call at the line', () => {
+  mockFile('/project/some-file.js', 'somethingDifferent()')
+  expect(
+    createErrorMessageWithContext({
+      originalMethod: 'model.findFirst',
+      callsite: mockCallsite('/project/some-file.js', 1, 1),
+      message: 'What a terrible failure!',
+    }),
+  ).toMatchInlineSnapshot(`
+
+    Invalid \`prisma.model.findFirst()\` invocation:
+
+
+    What a terrible failure!
+  `)
```

Comment:
I think it could happen if either code is bundlled and source map is incorrect, or users doing something weird re-assignments along the lines of:

```ts
let somethingDifferent = prisma.user.findFirst()
somethingDifferent()
```

Reply:
Do we want to print somethign like `(Invocation location could not be found)` instead? Or even make users create an issue? Or just remove the `:` maybe and one of the two empty lines?

Or actually just leave it as it is? :D

Reply:
Sorry, this PR got lost for quite some time. Not sure if you remember the context of this conversation but will reply anyway. I am leaning towards leaving it as is for following reasons:

- "Invocation line could not be found" might be confused for an actual error message
- There is hardly anything actionable for us: if it is a bundling/source map issue, then it is caused by a bundler and we can not do anything about it. If it is weird re-assignment from example above - everything works as expected, I can't imagine we will ever be able to detect that. So, suggesting to open github issue might also not make more sense.
- Error message and the method name is still present and it actuality, raw stack will be present as well. So, even though message might not look as nice, we are at least not using any information present in original exception.